src/NServiceBus.Core/Routing/RoutingSettingsExtensions.cs

```diff
@@ -1,17 +1,52 @@
 ﻿namespace NServiceBus
 {
+    using Routing;
+    using Transports;
+
     /// <summary>
     /// Configuration extensions for routing.
     /// </summary>
     public static class RoutingSettingsExtensions
     {
         /// <summary>
-        /// Controls the unicast routing.
+        /// Configures the routing.
         /// </summary>
-        public static RoutingSettings Routing(this EndpointConfiguration config)
+        public static RoutingSettings Routing(this TransportExtensions config)
         {
             Guard.AgainstNull(nameof(config), config);
             return new RoutingSettings(config.Settings);
-        }        
+        }
+
+        /// <summary>
+        /// Configures the routing.
+        /// </summary>
+        public static RoutingSettings<T> Routing<T>(this TransportExtensions<T> config) 
+            where T : TransportDefinition
+        {
+            Guard.AgainstNull(nameof(config), config);
+            return new RoutingSettings<T>(config.Settings);
+        }
+
+        /// <summary>
+        /// Sets a distribution strategy for a given endpoint.
+        /// </summary>
+        /// <param name="config">Config object.</param>
+        /// <param name="endpointName">The name of the logical endpoint the given strategy should apply to.</param>
+        /// <param name="distributionStrategy">The instance of a distribution strategy.</param>
+        public static void SetMessageDistributionStrategy<T>(this RoutingSettings<T> config, string endpointName, DistributionStrategy distributionStrategy) 
```

Comment:
Because it better expresses the intent. Plus it would make it easier to move MSMQ out of the core without touching the routing system.